test/unittest/middleware/ced/CedMiddlewareTests.cpp

```diff
@@ -0,0 +1,907 @@
+// Copyright 2019 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <uxr/agent/middleware/ced/CedMiddleware.hpp>
+
+#include <gtest/gtest.h>
+
+namespace eprosima {
+namespace uxr {
+namespace testing {
+
+class CedMiddlewareUnitTests : public ::testing::Test
+{
+public:
+    CedMiddlewareUnitTests() = default;
+    ~CedMiddlewareUnitTests() = default;
+
+protected:
+    CedMiddleware middleware_;
+};
+
+TEST_F(CedMiddlewareUnitTests, CreateParticipantByRef)
+{
+    std::string ref_one{"ref_one"};
```

Comment:
There are a lot of ways to initialize a string, but I never see this way :laughing:

Reply:
Yeah, that is a list initialization.

Reply:
Yeah, that is a list initialization.

Reply:
For any reason in special? there is any different behaviour to the "normal" initialization?

Reply:
It should be [more efficient](https://stackoverflow.com/questions/18222926/why-is-list-initialization-using-curly-braces-better-than-the-alternatives) than the normal constructor.